tests/test_encode_decode.py

```diff
@@ -1,12 +1,62 @@
+import base64
 import quopri
 import unittest
 
 from hypothesis import given, strategies as st
 
 
 class TestBase64(unittest.TestCase):
-    # TODO: https://docs.python.org/3/library/base64.html
-    pass
+    @given(payload=st.binary())
```

Comment:
I added `altchars` and `validate` arguments. 
From the docs for `altchars` - `Optional altchars should be a byte string of length 2 which specifies an alternative alphabet for the '+' and '/' characters.  This allows an application to e.g. generate url or filesystem safe Base64 strings.`

Based on this, I came up with this test.

```
@given(
        payload=st.binary(),
        altchars=(st.none() | st.binary(min_size=2, max_size=2)),
        validate=st.booleans(),
    )
    def test_b64_encode_decode_round_trip(self, payload, altchars, validate):
        x = base64.b64encode(payload, altchars=altchars)
        self.assertEqual(
            payload, base64.b64decode(x, altchars=altchars, validate=validate)
        )
```

But, I observe that this fails. Example - 
```
payload=b"\x00\x00\x00\x03\xd0",
altchars=b"\x009",
base64.b64encode(payload, altchars=altchars) = b"AAAAA9A="
```
So the `9` in the encoded string, gets decoded back to `/` (because of `altchars`)


I am not sure, if this is a bug. 

I looked up a lot of examples on the internet, and all of them use `altchars = b"_-"`

In that case the test case (as below), works completely fine.
```
@given(
        payload=st.binary(),
        altchars=(st.none() | st.just(b"_-")),
        validate=st.booleans(),
    )
    def test_b64_encode_decode_round_trip(self, payload, altchars, validate):
        x = base64.b64encode(payload, altchars=altchars)
        self.assertEqual(
            payload, base64.b64decode(x, altchars=altchars, validate=validate)
        )
```

How do we proceed?

Reply:
How should I proceed on this?